spec/feature_flagger/storage/redis_spec.rb

```diff
@@ -3,63 +3,103 @@
-  let(:key)   { 'foo' }
-  let(:value) { 'bar' }
-  let(:global_key) { 'released_features' }
+  let(:feature_key)   { 'account:email_marketing:whitelabel' }
+  let(:resource_id)   { '1' }
+  let(:resource_name) { 'account' }
+  let(:global_key)    { 'released_features' }
+  let(:resource_key) do
+    FeatureFlagger::Storage::RedisKeys.resource_key(
+      FeatureFlagger::Storage::Redis::RESOURCE_PREFIX,
+      resource_name,
+      resource_id,
+    )
+  end
 
   context do
     before do
       redis.flushdb
     end
 
-    describe '#has_value?' do
-      context 'value is stored for given key' do
-        before { redis.sadd(key, value) }
-        it { expect(storage).to have_value(key, value) }
+    describe '#has_values?' do
+      context 'resource_id is stored for given feature_key' do
+        before { redis.sadd(feature_key, resource_id) }
+        it { expect(storage).to have_value(feature_key, resource_id) }
+      end
+
+      context 'resource_id is not stored for given feature_key' do
+        it { expect(storage).not_to have_value(feature_key, resource_id) }
+      end
+    end
+
+    describe '#fetch_releases' do
+      context 'when there is no features under global structure' do
+        before do
+          storage.add(feature_key, resource_name, resource_id)
+        end
+
+        it 'returns related feature_keys' do
+          expect(storage.fetch_releases(resource_name, resource_id, global_key)).to match_array([feature_key])
+        end
       end
 
-      context 'value is not stored for given key' do
-        it { expect(storage).not_to have_value(key, value) }
+      context 'when there is features under global structure' do
+        before do
+          storage.add(feature_key, resource_name, resource_id)
+          storage.add_all(global_key, feature_key)
+        end
+
+        it 'returns related feature_keys' do
+          expect(storage.fetch_releases(resource_name, resource_id, global_key)).to match_array(feature_key)
+        end
       end
     end
 
     describe '#add' do
-      it 'adds the value to redis' do
-        storage.add(key, value)
-        expect(redis.sismember(key, value)).to be_truthy
+      it 'adds the resource_id to redis' do
+        storage.add(feature_key, resource_name, resource_id)
+
+        expect(redis.sismember(feature_key, resource_id)).to be_truthy
+        expect(redis.sismember(resource_key, feature_key)).to be_truthy
       end
     end
 
     describe '#add_all' do
-      it 'adds value to redis global key and clear key' do
-        storage.add_all(global_key, key)
-        expect(redis.sismember(global_key, key)).to be_truthy
-        expect(redis.sismember(key, value)).to be_falsey
+      it 'adds resource_id to redis global feature_key and clear key' do
+        storage.add_all(global_key, resource_id)
+        expect(redis.sismember(global_key, resource_id)).to be_truthy
+
+        expect(redis.sismember(feature_key, resource_id)).to be_falsey
```

Comment:
http://www.betterspecs.org/#single

Reply:
this can be discussed/done in another PR

Reply:
IMHO it's fine as it is. The single expectation tip is not well accepted by the community and will probably be removed from the guide.

https://github.com/betterspecs/betterspecs/issues/5
https://stackoverflow.com/questions/38437162/whats-the-difference-between-rspecs-subject-and-let-when-should-they-be-used/38459039#38459039

Reply:
Besides, the `expects` are co-dependent, we don't want to isolate then, in order the expect from line 71 be valid, it needs to `expect` from line 69 to pass.

Reply:
But I think this can be discussed in another issue, or PR. not related to this change, as long this is actually an opinionated style.